schema/attribute.h

```diff
@@ -79,6 +79,13 @@ class SCHEMA_EXPORT Attribute : public XmlElement
     class Private;
     Private *d;
 };
+
+bool operator==(const Attribute &lhs, const Attribute &rhs);
```

Comment:
Is there any reason for this to be a global function rather than a member function?

One downside of a global function is that it's missing SCHEMA_EXPORT.
The other downside is the duplication (or the casting that would be necessary in order to call common methods like an `operator==(const XmlElement&, const XmlElement&)`. Better use member functions so that one can delegate to the base class with the BaseClass::operator== syntax.

I think the benefit of global functions is the support for implicit conversions for both arguments but I'm not sure if it applies here.

Note: if you make it a member function, make it `const`, of course.

(This applies to all classes modified in this commit)